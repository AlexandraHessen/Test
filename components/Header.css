.Header .Info{
    display: flex;
    align-items: center;
    justify-content: space-around;
    padding-top: 10px;
}
.Header .Logo{
    font-family: 'Pattaya', sans-serif;
    letter-spacing: 2px;
    font-size: 2.5em;
    font-weight: bold;
    color: #fff;

}

.Header .ImgLogo{
    width: 60px;
}

.Header .MenuPages{
    display: flex;
    justify-content: center; 
    padding: 30px;
}

.Header .Contacts{

}

@media all and (max-width: 500px){
    .Header .Contacts{
        display: none;
    }

    .Header .Logo{
        font-size: 2.5em;
    }

    .Header .ImgLogo{
        width: 45px;
    }

}