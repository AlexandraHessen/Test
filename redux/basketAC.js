// ---------------- ActionCreator ----------------//
const ADD_PRODUCT = "ADD_PRODUCT"
const DEL_PRODUCT = "DEL_PRODUCT"
const CLEAR_BASKET = "CLEAR_BASKET" //пользователь сам очищает корзину при желании

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_product = function(productId, objProductInfo) {
    return {
      type: ADD_PRODUCT,
      productId: productId,
      objProductInfo: objProductInfo
    };
  }
  
  const del_product = function(productId) {
    return {
      type: DEL_PRODUCT,
      productId: productId,
    };
  }

  const clear_basket = function() {
    return {
      type: CLEAR_BASKET
    };
  }

export {
    add_product, ADD_PRODUCT,
    del_product, DEL_PRODUCT,
    clear_basket, CLEAR_BASKET,
  }