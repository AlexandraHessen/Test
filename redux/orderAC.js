// ---------------- ActionCreator ----------------//
const ADD_ORDER = "ADD_ORDER"

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_order = function(objCustomerInfo, objOrderInfo) {
    return {
      type: ADD_ORDER,
      objCustomerInfo: objCustomerInfo,
      objOrderInfo: objOrderInfo
    };
  }

export {
    add_order, ADD_ORDER,
  }