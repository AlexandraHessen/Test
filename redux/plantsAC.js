// ---------------- ActionCreator ----------------//
const PLANTS_LOADING='PLANTS_LOADING';
const PLANTS_ERROR='PLANTS_ERROR';
const PLANTS_SET='PLANTS_SET';
//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator
const plantsLoadingAC=function() {
  return {
    type: PLANTS_LOADING,
  };
}

const plantsErrorAC=function() {
  return {
    type: PLANTS_ERROR,
  };
}

const plantsSetAC=function(plants) {
  return {
    type: PLANTS_SET,
    plants:plants,
  };
}

export {
  plantsLoadingAC,PLANTS_LOADING,
  plantsErrorAC,PLANTS_ERROR,
  plantsSetAC,PLANTS_SET,
}
