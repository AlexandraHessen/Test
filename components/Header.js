"use strict";

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import Contacts from '../components/Contacts.js'
import MenuPages from '../pages/MenuPages';
import BasketShowCount from './BasketShowCount';
import './Header.css';


class Header extends React.Component{

    render(){
        return(
            <header className="Header">
                <div className="Info">
                    <div className="Logo">
                        <span>Plants</span>
                        <img src="/img/logo.png" className="ImgLogo"></img>
                    </div>
                    <div className='Contacts'>
                        <Contacts />
                    </div>
                    <BasketShowCount />
                </div>
                <div className="Navigation">
                    <MenuPages />
                </div>
            </header>
        )
    }
}

export default Header