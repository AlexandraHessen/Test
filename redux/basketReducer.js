// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_PRODUCT, DEL_PRODUCT, CLEAR_BASKET } from './basketAC'; //название action type


const initState={ // начальный state
    productsInBasket: {}
    //КЛЮЧИ productId, objProductInfo
    // 1: {code: 1, name: 'Замиокулькас'}

//   status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
//   data: null, //даные которые мы загрузили по сети

}

function basketReducer(state=initState,action) {
  switch (action.type) {

    case ADD_PRODUCT: {
      console.log('action:',action);
      console.log('state до обработки редьюсером:',state);
      let newState={...state, //старый state
        productsInBasket:{...state.productsInBasket, //заменить старые продукты на теже +
          // ПРИЛЕТИТ какой продукт (productid) изменить и насколько(addvalue)
          [action.productId]:action.objProductInfo //+
          //[action.counterid] - В нужном продукте 
          // state.productsInBasket[action.productid] - его старое значение 
          // action.addvalue - изменить на прилетевшее
        }
      };
      console.log('state после обработки редьюсером:',newState);
      return newState;
    }

    case DEL_PRODUCT: {
      let newState = {...state};
      delete newState.productsInBasket[action.productId];
      return newState;
    }

    case CLEAR_BASKET:{
      let newState = {...state};
      newState.productsInBasket={};
      return newState;
    }

    // case PLANTS_ERROR: {
    //   let newState={
    //     status:2,
    //     data:null,
    //   };
    //   return newState;
    // }

    // case PLANTS_SET: {
    //   let newState={
    //     status:3,
    //     data:action.plants, //загруженные данные
    //   };
    //   return newState;
    // }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default basketReducer;
