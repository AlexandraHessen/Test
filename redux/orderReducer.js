// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_ORDER } from './orderAC'; //название action type

const initState={ // начальный state
    orders: []
}

function orderReducer(state=initState,action) {
  switch (action.type) {
    case ADD_ORDER: {
      let newOrder=[...state.orders]
      newOrder.push( { objCustomerInfo: action.objCustomerInfo, products: action.objOrderInfo } );
      let newState={...state, orders: newOrder};

      return newState;
    }

              //обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default orderReducer;