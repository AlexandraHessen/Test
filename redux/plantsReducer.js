// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { PLANTS_LOADING, PLANTS_ERROR, PLANTS_SET } from './plantsAC'; //название action type


const initState={ // начальный state

  status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
  data: null, //даные которые мы загрузили по сети

}

function plantsReducer(state=initState,action) {
  switch (action.type) {

    case PLANTS_LOADING: {
      let newState={
        status:1,
        data:null,
      };
      return newState;
    }

    case PLANTS_ERROR: {
      let newState={
        status:2,
        data:null,
      };
      return newState;
    }

    case PLANTS_SET: {
      let newState={
        status:3,
        data:action.plants, //загруженные данные
      };
      return newState;
    }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default plantsReducer;
