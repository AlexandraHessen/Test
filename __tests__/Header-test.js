"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import Header from '../components/Header';
                    //импортируем файл который будем тестировать

let store=createStore(combinedReducer);

test('Header', () => {
                            // создаём тестовую версию компонента который будем тестировать
                            const component = renderer.create(
    <Provider store={store}>
        <BrowserRouter>
            <Header/>
        </BrowserRouter>
    </Provider>
    );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
