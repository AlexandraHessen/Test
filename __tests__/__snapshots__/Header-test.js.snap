// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<header
  className="Header"
>
  <div
    className="Info"
  >
    <div
      className="Logo"
    >
      <span>
        Plants
      </span>
      <img
        className="ImgLogo"
        src="/img/logo.png"
      />
    </div>
    <div
      className="Contacts"
    >
      <ul
        className="Contacts"
      >
        <li>
          +375 (29) 844-44-88
        </li>
        <li>
          +375 (44) 888-44-88
        </li>
        <li>
          Пн-Пт: 9:00-18:00
        </li>
        <li>
          г.Минск, пр.Победителей, 65
        </li>
      </ul>
    </div>
    <a
      aria-current={false}
      className="BasketShowCount"
      href="/basket"
      onClick={[Function]}
      style={undefined}
    >
      <button
        className="ButtonToBasket"
        type="button"
      >
        <i
          className="fas fa-shopping-cart"
        />
        <div
          className="CountProductsInBasket"
        >
          0
        </div>
      </button>
    </a>
  </div>
  <div
    className="Navigation"
  >
    <ul
      className="MenuPages"
    >
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/"
          onClick={[Function]}
          style={undefined}
        >
          Главная
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/catalog/1"
          onClick={[Function]}
          style={undefined}
        >
          Каталог
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/contacts"
          onClick={[Function]}
          style={undefined}
        >
          Контакты
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/basket"
          onClick={[Function]}
          style={undefined}
        >
          Корзина
        </a>
      </li>
    </ul>
  </div>
</header>
`;
